### PR TITLE
fixed #4571

### LINKED ISSUE #4571: [DB] Database file not created on boot
#### Current Behavior
```
➜  meshery git:(master) ✗ make run-fast
cd cmd; go mod tidy; \
	BUILD="v0.5.70" \
	PROVIDER_BASE_URLS="https://meshery.layer5.io" \
	PORT=9081 \
	DEBUG=true \
	ADAPTER_URLS="mesherylocal.layer5.io:10000 mesherylocal.layer5.io:10001 mesherylocal.layer5.io:10002 mesherylocal.layer5.io:10003 mesherylocal.layer5.io:10004 mesherylocal.layer5.io:10005 mesherylocal.layer5.io:10006 mesherylocal.layer5.io:10007 mesherylocal.layer5.io:10008 mesherylocal.layer5.io:10009 meshery-nginx-sm:10010" \
	go run main.go;
INFO[0000] Registered Meshery local Capabilities
INFO[0000] Meshery server current channel: Not Set
INFO[0000] Using '/Users/lee/.meshery/config' to store user data
INFO[0000] Using '/Users/lee/.kube' as the folder to look for kubeconfig file
INFO[0000] Log level: debug
ERRO[2021-11-05T13:35:40-05:00] msg[unable to open database file: no such file or directory]  app=meshery
FATA[0000] unable to open database file: no such file or directory
exit status 1
make: *** [run-fast] Error 1
➜  meshery git:(master) ✗ ls -l ~/.meshery/
total 88
drwxr-x---  4 lee  staff   128 Nov  5 12:52 bin
-rw-r--r--  1 lee  staff   431 Nov  5 12:35 config.yaml
-rw-r--r--  1 lee  staff  2337 Nov  5 12:51 consul.yaml
-rw-r--r--  1 lee  staff  5350 Nov  5 12:51 istio.yaml
-rw-r--r--  1 lee  staff     3 Nov  5 12:51 kubeconfig.yaml
-rw-r--r--  1 lee  staff  2084 Nov  5 12:52 kuma.yaml
-rw-r--r--  1 lee  staff  2229 Nov  5 12:51 linkerd.yaml
drwxr-xr-x  3 lee  staff    96 Nov  3 14:56 manifests
-rw-r--r--  1 lee  staff  2257 Nov  5 12:51 nginx.yaml
-rw-r--r--  1 lee  staff  2717 Nov  5 12:51 nsm.yaml
-rw-r--r--  1 lee  staff  2996 Nov  5 12:51 osm.yaml
drwxr-xr-x  3 lee  staff    96 Nov  5 12:54 provider
-rw-r--r--  1 lee  staff  2098 Nov  5 12:51 traefik.yaml
```

#### Environment

- **Host OS:** Mac
- **Platform:** Kubernetes
- **Meshery Server Version:** edge

---
#### Contributor [Guides](https://docs.meshery.io/project/contributing) and Resources
- 🛠 [Meshery Build & Release Strategy](https://docs.meshery.io/project/build-and-release)
- 📚 [Instructions for contributing to documentation](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#documentation-contribution-flow)
   - Meshery documentation [site](https://docs.meshery.io/) and [source](https://github.com/meshery/meshery/tree/master/docs)
- 🎨 Wireframes and designs for Meshery UI in [Figma](https://www.figma.com/file/SMP3zxOjZztdOLtgN4dS2W/Meshery-UI)
- 🙋🏾🙋🏼 Questions: [Layer5 Discussion Forum](https://discuss.layer5.io) and [Layer5 Community Slack](http://slack.layer5.io)

### PR DESCRIPTION
**Pull request by Revolyssup**
Signed-off-by: ashish <ashishjaitiwari15112000@gmail.com>

**Description**

This PR fixes #4571 

**Notes for Reviewers**

**[Signed commits](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

File: cmd/main.go [hunk 1]
```diff
@@ -94,8 +94,15 @@ func main() {
 		}
 		viper.SetDefault("USER_DATA_FOLDER", path.Join(home, ".meshery", "config"))
 	}
+	_, er := os.Stat(viper.GetString("USER_DATA_FOLDER"))
+	if os.IsNotExist(er) {
+		logrus.Infof("Creating directory for storing user data in %v", viper.GetString("USER_DATA_FOLDER"))
+		errDir := os.MkdirAll(viper.GetString("USER_DATA_FOLDER"), 0755)
```
Comment (utkarsh-pro) [inline cmd/main.go:100]: mkdirall would suffice, I don't think we have to stat before calling this.

Comment (Revolyssup) [reply cmd/main.go:100]: For running locally, i made the check

Comment (Revolyssup) [reply cmd/main.go:100]: Yeah I guess I can do that directly with Mkdirall

File: cmd/main.go [hunk 2]
```diff
@@ -94,8 +94,13 @@ func main() {
 		}
 		viper.SetDefault("USER_DATA_FOLDER", path.Join(home, ".meshery", "config"))
 	}
-	logrus.Infof("Using '%s' to store user data", viper.GetString("USER_DATA_FOLDER"))
 
+	errDir := os.MkdirAll(viper.GetString("USER_DATA_FOLDER"), 0755)
```
Comment (leecalcote) [inline cmd/main.go:98]: Do we need execution permission on the folder/files?

File: cmd/main.go [hunk 3]
```diff
@@ -94,8 +94,13 @@ func main() {
...
+	if errDir != nil {
+		logrus.Fatalf("unable to create the directory for storing user data at %v", viper.GetString("USER_DATA_FOLDER"))
```
Comment (leecalcote) [inline cmd/main.go:100]: Is this a MeshKit-style error?

**Review by leecalcote — APPROVED** [review]

**Review by leecalcote — APPROVED** [review]

**Pull request merged**